tools/src/test/java/org/neo4j/tools/txlog/CheckTxLogsTest.java

```diff
@@ -763,6 +767,66 @@ public void shouldReportNeoStoreInconsistenciesFromDifferentLogs() throws IOExce
         assertEquals( 13, currentRecord2.getNextProp() );
     }
 
+    @Test
+    public void shouldDetectAnInconsistentCheckPointPointingToALogFileGreaterThanMaxLogVersion() throws Exception
+    {
+        // given
+        File log = logFile( 1 );
+        writeCheckPoint( log, 2, LOG_HEADER_SIZE );
+
+        CapturingInconsistenciesHandler handler = new CapturingInconsistenciesHandler();
+        CheckTxLogs checker = new CheckTxLogs( System.out, fsRule.get() );
+
+        // when
+        checker.validateCheckPoints( new PhysicalLogFiles( storeDirectory, fsRule.get() ), handler );
+
+        // then
+        assertEquals( 1, handler.checkPointInconsistencies.size() );
+
+        assertEquals( 1, handler.checkPointInconsistencies.get( 0 ).logVersion );
+        assertEquals( new LogPosition( 2, LOG_HEADER_SIZE ), handler.checkPointInconsistencies.get( 0 ).logPosition );
+        assertNull( handler.checkPointInconsistencies.get( 0 ).size );
+    }
+
+    @Test
+    public void shouldDetectAnInconsistentCheckPointPointingToAByteOffsetNotInTheFile() throws Exception
+    {
+        // given
+        ensureLogExists( logFile( 1 ) );
+        writeCheckPoint( logFile( 2 ), 1, 42 );
+
+        CapturingInconsistenciesHandler handler = new CapturingInconsistenciesHandler();
+        CheckTxLogs checker = new CheckTxLogs( System.out, fsRule.get() );
+
+        // when
+        checker.validateCheckPoints( new PhysicalLogFiles( storeDirectory, fsRule.get() ), handler );
+
+        // then
+        assertEquals( 1, handler.checkPointInconsistencies.size() );
+
+        assertEquals( 2, handler.checkPointInconsistencies.get( 0 ).logVersion );
+        assertEquals( new LogPosition( 1, 42 ), handler.checkPointInconsistencies.get( 0 ).logPosition );
+        assertEquals( LOG_HEADER_SIZE, handler.checkPointInconsistencies.get( 0 ).size.longValue() );
+    }
+
+    @Test
+    public void shouldNotReportInconsistencyIfTheCheckPointAreValidOrTheyReferToPrunedLogs() throws Exception
```

Comment:
should it really be "or" here? I would not expect an inconsistency to be reported if the check point[s] are valid...